WordPress/src/main/java/org/wordpress/android/ui/uploads/MediaUploadManager.java

```diff
@@ -0,0 +1,299 @@
+    }
+
+    private MediaModel getNextMediaToUpload() {
+        if (!sPendingUploads.isEmpty()) {
+            return sPendingUploads.remove(0);
+        }
+        return null;
+    }
+
+    private void addUniqueMediaToQueue(MediaModel media) {
+        if (media != null) {
+            if (mediaAlreadyQueuedOrUploading(media)) {
+                return;
+            }
+
+            // no match found in queue
+            sPendingUploads.add(media);
+        }
+    }
+
+    private void cancelUpload(MediaModel oneUpload, boolean delete) {
+        if (oneUpload != null) {
+            SiteModel site = mSiteStore.getSiteByLocalId(oneUpload.getLocalSiteId());
+            if (site != null) {
+                dispatchCancelAction(oneUpload, site, delete);
+            } else {
+                AppLog.w(T.MEDIA, "MediaUploadManager > Unexpected state, site is null. "
+                        + "Skipping cancellation of this request.");
+            }
+        }
+    }
+
+    private void dispatchUploadAction(@NonNull final MediaModel media, @NonNull final SiteModel site) {
+        AppLog.i(T.MEDIA, "MediaUploadManager > Dispatching upload action for media with local id: "
+                + media.getId() + " and path: " + media.getFilePath());
+        sInProgressUploads.add(media);
+        mDispatcher.dispatch(MediaActionBuilder.newUpdateMediaAction(media));
+
+        MediaPayload payload = new MediaPayload(site, media);
+        mDispatcher.dispatch(MediaActionBuilder.newUploadMediaAction(payload));
+    }
+
+    private void dispatchCancelAction(@NonNull final MediaModel media, @NonNull final SiteModel site, boolean delete) {
+        AppLog.i(T.MEDIA, "MediaUploadManager > Dispatching cancel upload action for media with local id: "
+                + media.getId() + " and path: " + media.getFilePath());
+        CancelMediaPayload payload = new CancelMediaPayload(site, media, delete);
+        mDispatcher.dispatch(MediaActionBuilder.newCancelMediaUploadAction(payload));
+    }
+
+    private void notifyServiceIfUploadsComplete() {
+        if (sPendingUploads.isEmpty() && sInProgressUploads.isEmpty()) {
+            AppLog.i(T.MEDIA, "MediaUploadManager > Completed");
+        }
+    }
+
+    // App events
+
+    @SuppressWarnings("unused")
+    public void onEventMainThread(PostEvents.PostMediaCanceled event) {
+        if (event.post == null) {
+            return;
+        }
+        for (MediaModel inProgressUpload : sInProgressUploads) {
+            if (inProgressUpload.getLocalPostId() == event.post.getId()) {
+                cancelUpload(inProgressUpload, true);
+            }
+        }
+        for (MediaModel pendingUpload : sPendingUploads) {
+            if (pendingUpload.getLocalPostId() == event.post.getId()) {
+                cancelUpload(pendingUpload, true);
+            }
+        }
+    }
+
+    // FluxC events
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN, priority = 9)
```

Comment:
this is new - I only got some information about this here http://greenrobot.org/eventbus/documentation/priorities-and-event-cancellation/
Can we add a comment here explaining why and how / what we expect to happen by using the `priority` annotation here?